packages/core/src/test/amazonq/session/sessionState.test.ts

```diff
@@ -0,0 +1,189 @@
+            this.generatedFiles = newFileInfo
+        }
+        protected handleError(_messenger: any, _codegenResult: any): Error {
+            throw new Error('handleError called')
+        }
+    }
+
+    let fakeProxyClient: any
+    let testConfig: any
+    let fsMock: any
+    let messengerMock: any
+    let loggerMock: any
+    let testAction: any
+
+    beforeEach(async () => {
+        const ret = {
+            testworkspacefolder: {
+                uri: vscode.Uri.file('/path/to/testworkspacefolder'),
+                name: 'testworkspacefolder',
+                index: 0,
+            },
+        }
+        sinon.stub(workspaceUtils, 'getWorkspaceFoldersByPrefixes').returns(ret)
+
+        fakeProxyClient = {
+            getCodeGeneration: sinon.stub().resolves({
+                codeGenerationStatus: { status: 'Complete' },
+                codeGenerationRemainingIterationCount: 0,
+                codeGenerationTotalIterationCount: 1,
+            }),
+            exportResultArchive: sinon.stub(),
+        }
+
+        testConfig = {
+            conversationId: 'conv_test',
+            uploadId: 'upload_test',
+            workspaceRoots: ['/path/to/testworkspacefolder'],
+            proxyClient: fakeProxyClient,
+        }
+
+        fsMock = {
+            writeFile: sinon.stub().resolves(),
+            registerProvider: sinon.stub().resolves(),
+        }
+
+        messengerMock = { sendAnswer: sinon.spy() }
+
+        loggerMock = {
+            info: sinon.spy(),
+        }
+
+        testAction = {
+            fs: fsMock,
+            messenger: messengerMock,
+            logger: loggerMock,
+            tokenSource: {
+                token: {
+                    isCancellationRequested: false,
+                    onCancellationRequested: () => {},
+                },
+            },
+        }
+    })
+
+    afterEach(() => {
+        sinon.restore()
+    })
+
+    it('adds the log content to logger if present and excludes it from new files', async () => {
+        const logFileInfo = {
+            zipFilePath: RunCommandLogFileName,
+            fileContent: 'Log content',
+        }
+        const otherFile = { zipFilePath: 'other.ts', fileContent: 'other content' }
+        fakeProxyClient.exportResultArchive.resolves({
+            newFileContents: [logFileInfo, otherFile],
+            deletedFiles: [],
+            references: [],
+        })
+        const controlledLogger = {
+            info: sinon.spy(),
+            debug: sinon.spy(),
+            verbose: sinon.spy(),
+            warn: sinon.spy(),
+            error: sinon.spy(),
+            log: sinon.spy(),
+            setLogLevel: sinon.spy(),
+            setLogLevelToDefault: sinon.spy(),
+            logLevelEnabled: sinon.spy(),
+            getLogById: sinon.spy(),
+            sendToLog: sinon.spy(),
+        }
+        sinon.stub(loggerModule, 'getLogger').returns(controlledLogger)
```

Comment:
Do not mock the logger. Use `assertLogsContain` to test the actual contents of the actual logger.